api-spec/examples.md

```diff
@@ -3,53 +3,84 @@
-2. The parameters for a GET request must be escaped properly, making it more difficult to construct when using JSON parameters (such as intersect)
+2. The parameters for a GET request must be escaped properly, making it more difficult to construct when using JSON 
+parameters (such as intersect)
 
 ## Core examples
 
+Requests all the data in the collection that is in New Zealand:
+
 ```
-GET /collections/{name}/items?bbox=160.6,-55.95,-170,-25.89
+GET /stac/search?bbox=160.6,-55.95,-170,-25.89
 ```
 
-Requests all the data in the collection that is in New Zealand. The filtering is made to be compatible with the STAC API,
-and the two specs seek to share the general query and filtering patterns. The key difference is that a STAC search endpoint
-will do cross collection search. A typical WFS will have multiple collections, and each will just offer search for its particular
-collection.
+Request all the data in `mycollection` that is in New Zealand:
 
-Request all the data in mycollection that is in New Zealand:
+```
+GET /stac/search/items?bbox=160.6,-55.95,-170,-25.89&collections=mycollection
+```
+
+Request 100 results in `mycollection` from New Zealand:
 
 ```
-GET /collections/mycollection/items?bbox=160.6,-55.95,-170,-25.89
+GET /stac/search?bbox=160.6,-55.95,-170,-25.89&limit=100&collections=mycollection
 ```
 
-Request 100 results in mycollection from New Zealand:
+Request all the data in `mycollection` that is in New Zealand from January 1st, 2019:
 
 ```
-GET /collections/mycollection/items?bbox=160.6,-55.95,-170,-25.89&limit=100
+GET /stac/search?bbox=160.6,-55.95,-170,-25.89&time=2019-01-01&collections=mycollection
+```
+
+Request 100 results in `mycollection` or `yourcollection` that is in New Zealand from January 1st, 2019:
+
+```
+GET /stac/search?bbox=160.6,-55.95,-170,-25.89&time=2019-01-01&limit=100&collections=mycollection,yourcollection
 ```
 
-Request results 101-200 in mycollection from New Zealand:
+Request all the data in any collection that is in New Zealand from January 1st, 2019:
 
 ```
-GET /collections/mycollection/items?bbox=160.6,-55.95,-170,-25.89&page=2&limit=100
+GET /stac/search?bbox=160.6,-55.95,-170,-25.89&time=2019-01-01
 ```
 
-Request all the data in mycollection that is in New Zealand from January 1st, 2019:
+Request all the data in any collection that is in New Zealand and has the id `1` or `2`:
 
 ```
-GET /collections/mycollection/items?bbox=160.6,-55.95,-170,-25.89&time=2019-01-01
+GET /stac/search?bbox=160.6,-55.95,-170,-25.89&time=2019-01-01&ids=1,2
 ```
 
-Request all the data in mycollection from between January 1st and April 1st, 2019:
+Request 100 results in `mycollection` or `yourcollection` that is in New Zealand and has the id `1` or `2`:
 
 ```
-GET /collections/mycollection/items?time=2019-01-01/2019-04-01
+GET /stac/search?bbox=160.6,-55.95,-170,-25.89&time=2019-01-01&limit=100&collections=mycollection,yourcollection&ids=1,2
+```
+
+```
+POST /stac/search
+```
+
+Body:
```

Comment:
I'd prefer if we required this to be `multipart/form-data` instead of a JSON query body.

Reply:
Why?

Reply:
I prefer that because all of the values passed to the parameters are exactly the same in the GET and POST cases.  Otherwise, you have to define a separate query payload JSON schema, with similar (but not exact) mappings between values, for example, that `ids` is an array of strings rather than a comma-separated list of strings. With a JSON payload, we also get into the tricky situation when using the `query` extension where the value of one of the  attributes in the JSON is a string representing another query language, e.g., CQL or GraphQL, which can expose all sorts of bugs in JSON implementations (similarly to how our use of `:` in extension prefixes do)

I see more APIs that use this convention. for example, https://developer.twitter.com/en/docs/tweets/search/api-reference/enterprise-search#Methods

Reply:
That would need to be filed as a separate issue.  From the beginning we have used raw application/json.  This examples page is examples of the current spec.

Reply:
Ah right, will do that.

Reply:
Created a PR for some explicit language around supporting POST. #489